mitielib/include/mitie.h

```diff
@@ -804,6 +859,122 @@ extern "C"
+            - The returned object MUST BE FREED by a call to mitie_free().
+            - returns NULL if the object could not be created.
+            - calling mitie_text_categorizer_trainer_get_beta() on the returned pointer returns 0.5.
+              That is, the default beta value is 0.5.
+            - calling mitie_text_categorizer_trainer_get_num_threads() on the returned pointer returns
+              4.  That is, the default number of threads to use is 4.
+            - calling mitie_text_categorizer_trainer_size() on the returned pointer returns 0.
+              That is, initially there are no training instances in the trainer.
+    !*/
+
+    MITIE_EXPORT unsigned long mitie_text_categorizer_trainer_size (
+        const mitie_text_categorizer_trainer* trainer
+    );
+    /*!
+        requires
+            - trainer != NULL
+        ensures
+            - returns the number of training instances in the given trainer object.
+    !*/
+
+
+    MITIE_EXPORT void mitie_text_categorizer_trainer_set_beta (
+        mitie_text_categorizer_trainer* trainer,
+        double beta
+    );
+    /*!
+        requires
+            - trainer != NULL
+            - beta >= 0
+        ensures
+            - mitie_text_categorizer_trainer_get_beta(trainer) == beta
+    !*/
+
+    MITIE_EXPORT double mitie_text_categorizer_trainer_get_beta (
+        const mitie_text_categorizer_trainer* trainer
+    );
+    /*!
+        requires
+            - trainer != NULL
+        ensures
+            - returns the trainer's beta parameter.  This parameter controls the trade-off
+              between trying to avoid false alarms but also detecting everything.
+              Different values of beta have the following interpretations:
+              // TODO document interpretations of beta parameter
+    !*/
+
+    MITIE_EXPORT void mitie_text_categorizer_trainer_set_num_threads (
+        mitie_text_categorizer_trainer* trainer,
+        unsigned long num_threads
+    );
+    /*!
+        requires
+            - trainer != NULL
+        ensures
+            - mitie_text_categorizer_trainer_get_num_threads(trainer) == num_threads
+    !*/
+
+    MITIE_EXPORT unsigned long mitie_text_categorizer_trainer_get_num_threads (
+        const mitie_text_categorizer_trainer* trainer
+    );
+    /*!
+        requires
+            - trainer != NULL
+        ensures
+            - returns the number of threads the trainer will use when
+              mitie_train_text_categorizer() is called.  You should set this equal to
+              the number of available CPU cores for maximum training speed.
+    !*/
+
+    MITIE_EXPORT int mitie_add_text_categorizer_labeled_text (
+        mitie_text_categorizer_trainer* trainer,
+        const char** tokens,
+        const char* label
+    );
+    /*!
+        requires
+            - tokens != NULL
+            - label != NULL
+        ensures
+            - Adds the given training example to the trainer object.
```

Comment:
You should say something about what tokens is.  Like it's a sentence of words and that's the training example.  label should be mentioned in the text too.  Basically, what does this function do?  The complete answer to that needs to reference the arguments in some way.